src/main/java/seedu/task/model/task/DescContainsKeywordsPredicate.java

```diff
@@ -16,7 +16,8 @@ public DescContainsKeywordsPredicate(String keyphrase) {
 
     @Override
     public boolean test(Task task) {
-        return StringUtil.containsIgnoreCase(task.getDescription().value, keyphrase);
+        return StringUtil.containsIgnoreCase(task.getDescription().value, keyphrase)
+                && task.getDescription().getHasDescription();
```

Comment:
How about abstracting this into a method within Description?

Reply:
DescContainsKeywordsPredicate is itself an abstraction away from Description hence I think further abstraction would not do much good here.